### PR TITLE
Fix minor config bugs

### PR DESCRIPTION
**Pull request by jordlay**
NOTE: ANDY HAS REVIEWED FIRST 3 BUGS ALREADY. THIS REVIEW SHOULD FOCUS ON NFVISFROMSITE CHANGES.

NFVIFromSite Change:
- Each nfdv directly maps to an nfviFromSite object
- Added loop in template to have unique names and keys
- If all nfvi types are azure core, only have one nfviFromSite object
- 
image

After nsd publish, looks like this for multiple nfs:
{
    "nfvi1": {
        "name": "jl-testnfvis-nsd_NFVI1",
        "type": "AzureArcKubernetes"
    },
    "nfvi2": {
        "name": "jl-testnfvis-nsd_NFVI2",
        "type": "AzureArcKubernetes"
    }
}
---
- renamed nfdvName to nfdv:
- 
image

Below image shows 2 of the changes:
- additional comment above RET list
- additional comment above name
image

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/aosm/azext_aosm/tests/latest/integration_tests/test_nsd.py [hunk 1]
```diff
@@ -44,7 +44,7 @@
 
 CGV_DATA = {
     "ubuntu": {
-        "nfdvName": "exampleNfdvName",
+        "nfdv": "examplenfdv",
```
Comment (Cyclam) [inline src/aosm/azext_aosm/tests/latest/integration_tests/test_nsd.py:47]: Is this correct? I thought the confusion here was that this is actually the version string (e.g. 1.0.0)

Sometimes people look at tests to work out what's happening, so we should get our tests as accurate as possible.

Same comment for lines 63, 85 and 92.

Comment (jordlay) [reply src/aosm/azext_aosm/tests/latest/integration_tests/test_nsd.py]: youre right, changed

**Comment by Cyclam** [comment]
Looks like the NFVIs change has been tested for `build`. What about `publish` and then deploying SNS? We shouldn't merge to main until we're confident the new format is good.

File: src/aosm/azext_aosm/cli_handlers/onboarding_nsd_handler.py [hunk 2]
```diff
@@ -59,6 +59,7 @@ class OnboardingNSDCLIHandler(OnboardingBaseCLIHandler):
 
     config: OnboardingNSDInputConfig
     processors: list[AzureCoreArmBuildProcessor | NFDProcessor]
+    nfvi_types: list = []
```
[the remark this answers is not in the document]
Comment (Cyclam) [reply src/aosm/azext_aosm/cli_handlers/onboarding_nsd_handler.py:62]: I'm also slightly uneasy about setting this to an empty list here (i.e. outside of `__init__`, but maybe it's just a convention I'm not familiar with.  If static checks pass, I guess it's OK.

File: src/aosm/azext_aosm/cli_handlers/onboarding_nsd_handler.py [hunk 3]
```diff
@@ -234,13 +239,18 @@ def build_resource_bicep(self) -> BicepDefinitionElementBuilder:
             # List of NF RET names, for adding to required part of CGS
             nf_names.append(processor.name)
 
+
         template_path = get_template_path(
             NSD_TEMPLATE_FOLDER_NAME, NSD_DEFINITION_TEMPLATE_FILENAME
         )
 
+        # If all NF RETs nfvi_types are AzureCore, only make one nfviFromSite object
```
Comment (Cyclam) [inline src/aosm/azext_aosm/cli_handlers/onboarding_nsd_handler.py:247]: I'd argue this check should come immediately after the for loop, as it's related to the loop, whereas setting the `template_path` is not.

A minor point, and maybe there's a reason you wanted it here, but in general good to keep related operations together.

Comment (Cyclam) [reply src/aosm/azext_aosm/cli_handlers/onboarding_nsd_handler.py:247]: Also, can you explain _why_ we do this. Ideally, the code should be clear on the what, and the comment should explain the why.

File: src/aosm/azext_aosm/cli_handlers/onboarding_nsd_handler.py [hunk 4]
```diff
@@ -234,13 +239,18 @@ def build_resource_bicep(self) -> BicepDefinitionElementBuilder:
             # List of NF RET names, for adding to required part of CGS
             nf_names.append(processor.name)
 
+
         template_path = get_template_path(
             NSD_TEMPLATE_FOLDER_NAME, NSD_DEFINITION_TEMPLATE_FILENAME
         )
 
+        # If all NF RETs nfvi_types are AzureCore, only make one nfviFromSite object
+        if len(set(self.nfvi_types)) == 1 and self.nfvi_types[0] == 'AzureCore':
```
Comment (Cyclam) [inline src/aosm/azext_aosm/cli_handlers/onboarding_nsd_handler.py:248]: Any reason not just to do this?:

```
        if all(nfvi_type == "AzureCore" for nfvi_type in self.nfvi_types):
            self.nfvi_types = ["AzureCore"]
```

**Review by Cyclam — CHANGES_REQUESTED** [review]

File: src/aosm/azext_aosm/cli_handlers/onboarding_nsd_handler.py (hunk 4 above)
Comment (jordlay) [reply src/aosm/azext_aosm/cli_handlers/onboarding_nsd_handler.py:248]: just not how i did it, changed

File: src/aosm/azext_aosm/cli_handlers/onboarding_nsd_handler.py [hunk 5]
```diff
@@ -6,7 +6,7 @@
 
 import json
 from pathlib import Path
-from typing import Optional
+from typing import Literal, Optional
```
Comment (Cyclam) [inline src/aosm/azext_aosm/cli_handlers/onboarding_nsd_handler.py:9]: Is this used?

File: src/aosm/azext_aosm/cli_handlers/onboarding_nsd_handler.py [hunk 6]
```diff
@@ -46,7 +46,7 @@
 from azext_aosm.definition_folder.builder.local_file_builder import LocalFileBuilder
 from azext_aosm.inputs.arm_template_input import ArmTemplateInput
 from azext_aosm.inputs.nfd_input import NFDInput
-from azext_aosm.vendored_sdks.models import NetworkFunctionDefinitionVersion
+from azext_aosm.vendored_sdks.models import NetworkFunctionDefinitionVersion, NFVIType
```
Comment (Cyclam) [inline src/aosm/azext_aosm/cli_handlers/onboarding_nsd_handler.py]: Is `NFVIType` used?

**Review by Cyclam — CHANGES_REQUESTED** [review]

File: src/aosm/azext_aosm/cli_handlers/onboarding_nsd_handler.py (hunk 5 above)
Comment (jordlay) [reply src/aosm/azext_aosm/cli_handlers/onboarding_nsd_handler.py:9]: yes both are used

**Review by Cyclam — APPROVED** [review]
LGTM

**Pull request merged**